### PR TITLE
Pal engine2

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

**Comment by ethteck** [comment]
thanks pal

File: src/msg.c [hunk 1]
```diff
@@ -436,7 +451,7 @@ s32 _update_message(MessagePrintState* printer) {
                     }
                 case MSG_WINDOW_STATE_SCROLLING_BACK:
                     printer->scrollingTime++;
-                    if (printer->scrollingTime >= 5) {
```
Comment (ethteck) [inline src/msg.c:439]: the cod is different here

File: ver/pal/asm/nonmatchings/msg/dma_load_msg.s [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+/* BA6A8 80124358 24850004 */   addiu     $a1, $a0, 0x4
+/* BA6AC 8012435C 27A60010 */  addiu      $a2, $sp, 0x10
+/* BA6B0 80124360 3210FFFF */  andi       $s0, $s0, 0xFFFF
+/* BA6B4 80124364 8FA20010 */  lw         $v0, 0x10($sp)
+/* BA6B8 80124368 00108080 */  sll        $s0, $s0, 2
+/* BA6BC 8012436C 02221021 */  addu       $v0, $s1, $v0
+/* BA6C0 80124370 00502021 */  addu       $a0, $v0, $s0
+/* BA6C4 80124374 0C00A5AF */  jal        dma_copy
+/* BA6C8 80124378 24850008 */   addiu     $a1, $a0, 0x8
+/* BA6CC 8012437C 0240302D */  daddu      $a2, $s2, $zero
```
Comment (ethteck) [inline ver/pal/asm/nonmatchings/msg/dma_load_msg.s:33]: dad

File: src/msg.c [hunk 3]
```diff
@@ -502,7 +517,7 @@ s32 _update_message(MessagePrintState* printer) {
                     (!(printer->stateFlags & (MSG_STATE_FLAG_10 | MSG_STATE_FLAG_4)) &&
                     (gGameStatusPtr->curButtons[0] & BUTTON_A)))
                 {
-                    printer->curLinePos += 6;
+                    printer->curLinePos += 7;
```
Comment (marijnvdwerf) [inline src/msg.c:520]: Is this X or Y?

Comment (ethteck) [reply src/msg.c:520]: Yex. Why?

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 550 functions (+5.95%) on `pal`.
src/hud_element.c:672: warning: comparison of distinct pointer types lacks a cast

**Review by nanaian — APPROVED** [review]

**Comment by ethteck** [comment]
@marijnvdwerf could you fix the warning, and then we merj

**Comment by marijnvdwerf** [comment]
No, because I don't understand the warning, and I don't know what `D_80200000` should actually be.

**Comment by ethteck** [comment]
Ah, okay. I'll fix it then

**Pull request merged**